packages/front-end/services/search.tsx

```diff
@@ -143,13 +144,28 @@ export function search<T>(index: IndexedObject<T>[], q: string): T[] {
 export function useSort<T>(
   items: T[],
   defaultField: string,
-  defaultDir: number = 1,
+  defaultDir: number,
+  fieldName: string,
   compFunctions?: Record<string, (a: T, b: T) => number>
 ) {
+  const [direction, setDirection] = useLocalStorage(
+    `${fieldName}:sort-dir`,
+    defaultDir
+  );
   const [sort, setSort] = useState({
     field: defaultField,
-    dir: defaultDir,
+    dir: direction,
   });
```

Comment:
I think it makes sense to persist the entire sort object (field + direction) in localStorage instead of just the direction.  That would simplify the code a lot too.  Basically, just need to swap the existing `useState` with `useLocalStorage` and you don't need a separate `updateSort` function to keep things in sync.